.github/workflows/release.yaml

```diff
@@ -1,63 +1,78 @@
+        required: true
+        default: 'master'
 
 jobs:
-  push-to-release-branch:
-    name: Push source to release branch
+  trigger-and-wait-for-mlrun-image-building:
+    name: Trigger build workflow in mlrun/mlrun and wait to finish
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v2
-      with:
-        # we can't use the default token coming with the action cause GH prevents from create or updating workflows
-        # which might be part of the release contents
-        token: ${{ secrets.RELEASE_GITHUB_ACCESS_TOKEN }}
-    - name: Push to release branch
-      run: |
-        export tag=$(echo ${GITHUB_REF#refs/tags/})
-        git config user.name "mlrun-iguazio"
-        git config user.email "mlrun.iguazio@gmail.com"
-        MLRUN_GIT_ORG=${{ github.repository_owner }} MLRUN_VERSION=$tag make release
-
+      - uses: convictional/trigger-workflow-and-wait@v1.6.3
+        with:
+          owner: mlrun
+          repo: mlrun
+          github_token: ${{ secrets.RELEASE_GITHUB_ACCESS_TOKEN }}
+          workflow_file_name: build.yaml
+          ref: ${{ github.event.inputs.branch }}
+          wait_interval: 60
+          client_payload: '{"docker_registries": "ghcr.io/,quay.io/,registry.hub.docker.com/", "version": "${{ github.event.inputs.version }}"}'
 
-  trigger-build-workflow:
-    # When the build workflow is triggered for a commit to development it builds the images and (docker) tag them with
-    # <latest-released-version>-<commit-hash>.
-    # When the periodic system tests workflow runs, it gets the latest commit from development, and searches for the
-    # relevant tag.
-    # Therefore, when a new version is released, any system test that will run after it, and before a new commit to
-    # development was done, will search for a docker tag that doesn't exist.
-    # In order to solve that we trigger the build workflow on the development branch when we release a new version
-    name: Trigger build workflow on development branch
+  trigger-and-wait-for-ui-image-building:
+    name: Trigger build workflow in mlrun/ui and wait to finish
     runs-on: ubuntu-latest
 
     steps:
+      - uses: convictional/trigger-workflow-and-wait@v1.6.3
+        with:
+          owner: mlrun
+          repo: ui
+          github_token: ${{ secrets.RELEASE_GITHUB_ACCESS_TOKEN }}
+          workflow_file_name: build.yaml
+          ref: ${{ github.event.inputs.branch }}
+          wait_interval: 60
+          client_payload: '{"docker_registries": "ghcr.io/,quay.io/,registry.hub.docker.com/", "version": "${{ github.event.inputs.version }}"}'
+
+  publish-to-pypi:
+    name: Publish package to pypi
+    runs-on: ubuntu-latest
+    needs: [trigger-and-wait-for-ui-image-building, trigger-and-wait-for-mlrun-image-building]
+    steps:
+      - name: Set up python ${{ matrix.python-version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.7
```

Comment:
Where matrix.python-version is coming from? What about 3.8?

Reply:
We're releasing not testing, no sense in releasing from different versions...
The matrix thing is just a copy pasting mistake, removed